source/custom/xnetsocket_converters.h

```diff
@@ -149,5 +149,68 @@ inline TimeValHolder convert_from_grpc(const pb_::Duration& input)
 {
   return TimeValHolder(input);
 }
+
+struct SockOptDataHolder {
+  SockOptData opt_data;
+  int32_t data_int;
```

Comment:
Can you change this to a `union` like `SocAddrHolder` (now `SockAddrInputConverter`)? Or make an argument for why not? You would need a switch case in the constructor because, like with `oneof`s, you need to only assign one value. But it makes it clearer that only one case is a "live" value.

Reply:
I tried but I think it was having trouble having a string in a union. I tried messing with it a little with a destructor in the case the string is set to destruct it but I think it came down to the compiler not liking the string as one of the data members.

Reply:
Ah, makes sense. Yeah, leave it then. I think `std::variant` would be best but we can't yet use that on nilrt.